file_optical.c

```diff
@@ -168,10 +168,6 @@ static void fbo_handler_init(struct fbo_handler *h, struct tcmu_device *dev,
 
 static void fbo_handler_destroy(struct fbo_handler *h)
 {
-	if (h->thr) {
-		pthread_kill(h->thr, SIGINT);
-		pthread_join(h->thr, NULL);
-	}
 	pthread_cond_destroy(&h->cond);
```

Comment:
We dont need these anymore because cancel_thread is called and when you call dev->handler->removed(dev); that also calls cancel_thread